text/0000-int-guidelines.md

```diff
@@ -0,0 +1,164 @@
+- Start Date: 2015-01-26
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes usage guidelines for the various integer types.
+
+# Motivation
+
+The goal of this RFC is to help people decide what integer types to use when they need
+to make a decision for a new API.
+
+It builds on [https://github.com/rust-lang/rfcs/pull/560](the integer overflow RFC),
+which provides debug-time assertions for overflow and underflow. One of the goals
+of this RFC is to provide guidance for when to use unsigned types, and these
+assertions affect the traditional tradeoffs about unsigned types, which are described
+under the detailed design below.
+
+It also draws inspiration from the [Google C++ Style Guide](http://google-styleguide.googlecode.com/svn/trunk/cppguide.html#Integer_Types)
```

Comment:
@thestinger Well, some Googlers might think along those lines but I've never heard anyone at Google say that. Also to commit C++ code, a programmer has to pass a "C++ readability review" or get commit approvals from someone who has. (Ditto in any other language.)

The key rationale behind the guidelines is to rule out some gotcha's, make the code more familiar, and avoid code churn due to minor choices.

This particular case is most likely a reaction to bugs in production.

Indeed, a Rust iterator is a better way to avoid the example count-down bug.